src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/ILToLLVMImporter.cs

```diff
@@ -4035,7 +4037,15 @@ private void ImportConvert(WellKnownType wellKnownType, bool checkOverflow, bool
             }
             else
             {
-                LLVMValueRef converted = CastIfNecessary(loadedValue, GetLLVMTypeForTypeDesc(destType), value.Name(), wellKnownType == WellKnownType.UInt64 /* unsigned is always false, so check for the type explicitly */);
+                TypeDesc sourceType = value.Type;
+
+                LLVMValueRef converted = CastIfNecessary(loadedValue, GetLLVMTypeForTypeDesc(destType), value.Name(),
+                    wellKnownType == WellKnownType.UInt64 /* unsigned is always false, so check for the type explicitly */
+                    || sourceType.IsWellKnownType(WellKnownType.Byte)
+                    || sourceType.IsWellKnownType(WellKnownType.UInt16)
+                    || sourceType.IsWellKnownType(WellKnownType.UInt32)
+                    || sourceType.IsWellKnownType(WellKnownType.UInt64)
+                    );
```

Comment:
How does the ILImporter stack work -- does it keep the precise types?

If it does, it does not look correct to look at them. E. g.
```
ldobj int64
conv.r.un
```
Should still be treated as an unsigned cast.

> `/* unsigned is always false, so check for the type explicitly */`

Is that true? I see it should be `true` for `conv.r.un`.

Reply:
Hmm, no that's not true, let me try to clean this up.

Reply:
There 2 reported problems, using a byte as an index and casting a byte to a float (for an alpha channel).  But looking at the code I don't see why I can't just pass the right value from the op code.